File: leetCode/263.ugly-number.cpp
/*
 * @lc app=leetcode id=263 lang=cpp
 *
 * [263] Ugly Number
 */

// @lc code=start
class Solution
{
public:
    bool isUgly(int n)
    {
        int div = 2;
        while (n > 1)
        {
            if (n % div == 0)
            {
                if (div != 2 && div != 3 && div != 5)
                    return false;
                n /= div;
            }
            else
                div++;
        }
        return true;
    }
};
// @lc code=end

File: leetCode/263.ugly-number.js
/*
 * @lc app=leetcode id=263 lang=javascript
 *
 * [263] Ugly Number
 */

// @lc code=start
/**
 * @param {number} n
 * @return {boolean}
 */
var isUgly = function (n) {
  if (n <= 0) return false;
  if (n === 1) return true;

  while (n % 2 === 0) n /= 2;

  while (n % 3 === 0) n /= 3;

  while (n % 5 === 0) n /= 5;

  return n === 1;
};
// @lc code=end

